pallets/settlement/src/benchmarking.rs

```diff
@@ -216,33 +231,345 @@ fn emulate_add_instruction<T: Trait>(
+) {
+    pallet_compliance_manager::Module::<T>::add_default_trusted_claim_issuer(
+        origin.into(),
+        ticker,
+        issuer,
+    )
+    .expect("Default trusted claim issuer cannot be added");
+}
+
+pub fn create_condition<T: Trait>(
+    c_count: u32,
+    t_count: u32,
+    ticker: Ticker,
+    token_owner: RawOrigin<T::AccountId>,
+) -> Vec<Condition> {
+    let condition_type = get_condition_type::<T>(&c_count, Scope::Ticker(ticker));
+    let trusted_issuers = make_issuers::<T>(t_count);
+    trusted_issuers.clone().into_iter().for_each(|issuer| {
+        let t_issuer = <pallet_compliance_manager::Module<T>>::trusted_claim_issuer(ticker);
+        if !t_issuer.contains(&issuer) {
+            add_trusted_issuer::<T>(token_owner.clone().into(), ticker, issuer);
+        }
+    });
+    vec![Condition::new(condition_type, trusted_issuers)]
+}
+
+pub fn get_condition_type<T: Trait>(condition_count: &u32, scope: Scope) -> ConditionType {
+    let target_identity = UserBuilder::<T>::default()
+        .generate_did()
+        .build("TargetIdentity")
+        .did();
+    if (2..8).contains(condition_count) {
+        ConditionType::IsPresent(Claim::Affiliate(scope))
+    } else if (9..18).contains(condition_count) {
+        ConditionType::IsAbsent(Claim::KnowYourCustomer(scope))
+    } else if (19..27).contains(condition_count) {
+        ConditionType::IsAnyOf(vec![
+            Claim::Affiliate(scope.clone()),
+            Claim::BuyLockup(scope.clone()),
+            Claim::SellLockup(scope.clone()),
+        ])
+    } else if (28..36).contains(condition_count) {
+        ConditionType::IsNoneOf(vec![
+            Claim::Accredited(scope.clone()),
+            Claim::Exempted(scope.clone()),
+            Claim::Jurisdiction(CountryCode::AF, scope.clone()),
+        ])
+    } else {
+        ConditionType::IsIdentity(TargetIdentity::Specific(target_identity))
+    }
+}
+
+fn compliance_setup<T: Trait>(
+    max_condition: u32,
+    max_trusted_issuer: u32,
+    ticker: Ticker,
+    origin: RawOrigin<T::AccountId>,
+    from_did: IdentityId,
+    to_did: IdentityId,
+    trusted_issuer: TrustedIssuer,
+    t_issuer_origin: RawOrigin<T::AccountId>,
+) {
+    // Add investor uniqueness claim.
+    add_investor_uniqueness_claim::<T>(from_did, ticker);
+    add_investor_uniqueness_claim::<T>(to_did, ticker);
+    // Add trusted issuer.
+    add_trusted_issuer::<T>(origin.clone(), ticker, trusted_issuer.clone());
+
+    let claim_to_pass = Claim::Accredited(Scope::Ticker(ticker));
+    for i in 0..max_condition {
+        let (s_cond, r_cond) = if i == (max_condition - 1) {
+            let cond = vec![Condition::new(
+                ConditionType::IsPresent(claim_to_pass.clone()),
+                vec![trusted_issuer.clone()],
+            )];
+            (cond.clone(), cond)
+        } else {
+            let cond = create_condition::<T>(i, max_trusted_issuer, ticker, origin.clone());
+            (cond.clone(), cond)
```

Comment:
These will generate the same conditions multiple times and hence the reads during execution will be same, cached reads. This is not the worse case.

Reply:
It will not generate the same condition anymore as the scope is different now.

Reply:
How? The scope is `Scope::Ticker(ticker)` always and `ticker` is constant.

Please see the above comment, there's no need to over-engineer here. Let's just stick with 1 rule with n conditions.

The ideology behind benchmarks is not to test every scenario. It's to test a single scenario multiple times, reliably.